File: Boxes/CodePal/Java/Boxes.java
import java.util.Scanner;

public class BoxPacker {
    
    /**
     * This function takes in the weight capacity of boxes and a list of item weights, and returns the minimum
     * number of boxes required to pack all the items.
     * 
     * @param capacity The weight capacity of each box
     * @param items An array of integers representing the weight of each item
     * @return The minimum number of boxes required to pack all the items
     */
    public static int packBoxes(int capacity, int[] items) {
        int numBoxes = 0;
        int currentBoxWeight = 0;
        
        for (int i = 0; i < items.length; i++) {
            if (currentBoxWeight + items[i] > capacity) {
                // If the current box is full, start a new box
                numBoxes++;
                currentBoxWeight = 0;
            }
            currentBoxWeight += items[i];
        }
        
        // If there are any remaining items, add another box
        if (currentBoxWeight > 0) {
            numBoxes++;
        }
        
        return numBoxes;
    }
    
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        // Read in the weight capacity of the boxes
        int capacity = scanner.nextInt();
        
        // Read in the number of items
        int numItems = scanner.nextInt();
        
        // Read in the weights of each item
        int[] items = new int[numItems];
        for (int i = 0; i < numItems; i++) {
            items[i] = scanner.nextInt();
        }
        
        // Calculate the minimum number of boxes required
        int numBoxes = packBoxes(capacity, items);
        
        // Print the result
        System.out.println(numBoxes);
    }
}

File: Boxes/CodePal/Python/Boxes.py
def main():
    try:
    
        weight_capacity = int(input())
        num_items = int(input())
        items = []
        for i in range(num_items):
            items.append(int(input()))

        
        # Check if weight capacity is an integer
        if not isinstance(weight_capacity, int):
            raise TypeError("Weight capacity must be an integer")
        
        # Check if items is a list of integers
        if not all(isinstance(item, int) for item in items):
            raise TypeError("Items must be a list of integers")
        
        # Sort the items in descending order
        items.sort(reverse=True)
        
        # Initialize variables
        num_boxes = 0
        current_box_weight = 0
        
        # Pack the items into boxes
        for item in items:
            if current_box_weight + item <= weight_capacity:
                current_box_weight += item
            else:
                num_boxes += 1
                current_box_weight = item
        
        # Add the last box if it's not empty
        if current_box_weight > 0:
            num_boxes += 1
        
        print(num_boxes)
    
    except TypeError as e:
        # Log the error
        print(f"Error: {e}")
        return 0
    

if __name__ == "__main__":
    main()
    